pkg/beacon/gjkr/message_filter_test.go

```diff
@@ -121,20 +127,20 @@ func TestFilterReconstructingMember(t *testing.T) {
 
 	member.MarkInactiveMembers(messages)
 
-	assertAcceptsFrom(member, 44, t) // should accept from self
-	assertAcceptsFrom(member, 41, t)
-	assertNotAcceptFrom(member, 42, t)
-	assertNotAcceptFrom(member, 43, t)
+	assertAcceptsFrom(member.memberCore, 44, t) // should accept from self
+	assertAcceptsFrom(member.memberCore, 41, t)
+	assertNotAcceptFrom(member.memberCore, 42, t)
+	assertNotAcceptFrom(member.memberCore, 43, t)
 }
 
-func assertAcceptsFrom(member group.MessageFiltering, senderID group.MemberIndex, t *testing.T) {
-	if !member.IsSenderAccepted(senderID) {
+func assertAcceptsFrom(member *memberCore, senderID group.MemberIndex, t *testing.T) {
+	if !member.group.IsOperating(senderID) {
 		t.Errorf("member should accept messages from [%v]", senderID)
 	}
 }
 
-func assertNotAcceptFrom(member group.MessageFiltering, senderID group.MemberIndex, t *testing.T) {
-	if member.IsSenderAccepted(senderID) {
+func assertNotAcceptFrom(member *memberCore, senderID group.MemberIndex, t *testing.T) {
+	if member.group.IsOperating(senderID) {
 		t.Errorf("member should not accept messages from [%v]", senderID)
 	}
 }
```

Comment:
I know `IsOperating` was what's been called under the hood but... I think it was wrong and if we want to make sure the changes in this PR did not break anything or did not waive any protection, we need to improve unit tests in this file.

`group.IsOperating` is implemented in `pkg/protocol/group/group.go` and should be tested in `pkg/protocol/group/group_test.go`. This is easy to add there with the existing tests.

Here, we should assert the result of `shouldAcceptMessage`. This way, we'll test that both marking as inactive works as designed but also if "from self" is correctly handled. We should also add tests for not matching public key. I do not think we do have tests for it elsewhere.

Reply:
Regarding `group.IsOperating`, I think this is covered by the `TestOperatingMembers` from `pkg/protocol/group/group_test.go`

Regarding `shouldAcceptMessage`, see b7912548